### PR TITLE
Costing for WT3 chlorination unit

### PR DESCRIPTION
**Pull request by andrewlee94**
## Changes proposed in this PR:
- Generalizes core functions in zero-order costing.py
- Adds costing method for chlorination unit

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by adam-a-a — DISMISSED** [review]
That was fast! LGTM

File: watertap/core/zero_order_costing.py [hunk 1]
```diff
@@ -314,81 +354,138 @@ def cost_chemical_addition(blk):
...
+            blk.unit_model.properties_in[t0].flow_vol /
+            (pyo.units.m**3/pyo.units.hour),
+            to_units=pyo.units.dimensionless)
+        ln_D = pyo.units.convert(
+            blk.unit_model.chlorine_dose[t0] / (pyo.units.mg/pyo.units.liter),
+            to_units=pyo.units.dimensionless)
+
+        expr = pyo.units.convert(
+            A*ln_Q + B*ln_D + C*ln_Q*ln_D,
+            to_units=blk.config.flowsheet_costing_block.base_currency)
```
Comment (adam-a-a) [inline watertap/core/zero_order_costing.py]: Actually, one retroactive question: where are you taking the log of these values (or why aren't you)?

Comment (andrewlee94) [reply watertap/core/zero_order_costing.py]: Oops, for got to add the log part. Fixing that now (along with the test failures).

**Review by adam-a-a — APPROVED** [review]
LGTM

**Review by kurbansitterley — APPROVED** [review]
LGTM

**Review by luohezhiming — APPROVED** [review]

**Pull request merged**